assets/js/util.js

```diff
@@ -242,8 +243,19 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function callApy(options, endpoint) {
+    return $.ajax(Object.assign({
+        url: config.APY_URL + endpoint,
+        beforeSend: ajaxSend,
+        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
+        dataType: 'json',
+        type: (options.data.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET'
```

Comment:
>  The most ideal solution is find a (simple) way to actually construct the URL and check.

&

> The URL length includes those data parameter key value pairs.

Okay, so the most ideal way would be:
1. Sum up the lengths of keys and values of all the params in `requestOptions` object.
2. Construct the URL similar to something that this snippet from `persistChoices` function generates?

```
        $.each(urlParamNames, function () {
            var urlParam = getURLParam(this);
            if(urlParam) {
                urlParams.push(this + '=' + encodeURIComponent(urlParam));
            }
        });

//

        else if(getURLParam('q' + qName).length > 0) {
            urlParams.push('q' + qName + '=' + getURLParam('q' + qName));
        }

        var newURL = window.location.pathname + (urlParams.length > 0 ? '?' + urlParams.join('&') : '') + hash;


```